metrics-datadog/src/main/java/org/coursera/metrics/datadog/DatadogReporter.java

```diff
@@ -73,6 +76,15 @@ public void report(SortedMap<String, Gauge> gauges,
                      SortedMap<String, Timer> timers) {
     final long timestamp = clock.getTime() / 1000;
 
+    if (tagsCallback != null) {
+      List<String> dynamicTags = tagsCallback.getTags();
+      if (dynamicTags != null && ! dynamicTags.isEmpty()) {
+        List<String> newTags = TagsMerger.mergeTags(tags, tagsCallback.getTags());
+        tags.clear();
+        tags.addAll(newTags);
```

Comment:
This logic seems to accumulate tags on every reporter run. I'm not sure this is the behavior we want. It could confuse users in that "stale" tags may stick around for the lifespan of the application. 

I think the contract should be that `DynamicTagCallback` always returns the full set of tags relevant at any point in time.  Hence, for each reporter run, we should take `tags` and merge in the results of a call to `dynamicTags.getTags()` into a new `List`. The original `tags` `List` should remain unchanged (it should be immutable) and we'll re-merge it on the next run. Does that make sense?